webfrontend/src/components/auth/register/register.tsx

```diff
@@ -0,0 +1,271 @@
+
+        fetch('/register_async', postRequestContent({
+            username: username,
+            password: password,
+            name: name,
+        }))
+            .then(res => {
+                if (res.status !== 200) {
+                    triggerErrorMessage();
+                    return;
+                }
+                res.text().then(res => {
+                    if (res !== 'account created') {
+                        setBottomErrorMessage("Account creation failed. Please Try again.")
+                    } else {
+                        dispatch(login({
+                            username: username,
+                        }));
+                    }
+                });
+            })
+    }, [username, password, password2, name, dispatch]);
+
+    return (
+    <div className="authentication-form-container">
+        <form
+            autoComplete="off"
+            action="{% url 'register_async' %}"
+            method="POST"
+            id="register-form"
+            className="authentication-form needs-validation"
+            noValidate
+            onSubmit={handleSubmit}
+        >
+            <div className="m-2">
+                <label htmlFor="name" className="form-label">Name <strong className="asterisk">*</strong>
+                </label>
+                <input
+                    type="text"
+                    name="name"
+                    id="name"
+                    className="form-control"
+                    autoFocus={true}
+                    value={name}
+                    spellCheck="false"
+                    onChange={e => {
+                        setName(e.target.value);
+                    }}
+                />
+                <div className="text-danger"><small>{nameWarning}</small></div>
+            </div>
+            <div className="m-2">
+            <label htmlFor="username" className="form-label">Username <strong className="asterisk">*</strong>
+            </label>
+            <input
+                type="text"
+                name="username"
+                id="username"
+                className="form-control"
+                value={username}
+                spellCheck="false"
+                onChange={e => {
+                    setUsername(e.target.value);
+                }}
+            />
+            <div className="text-danger"><small>{usernameWarning}</small></div>
+            </div>
+            <div className="m-2">
+            <label htmlFor="password" className="form-label">Password <strong className="asterisk">*</strong>
+            </label>
+            <input
+                type="password"
+                name="password"
+                id="password"
+                className="form-control"
+                value={password}
+                onChange={e => {
+                    setPassword(e.target.value);
+                }}
+            />
+            <div className="text-danger"><small>{passwordWarning}</small></div>
```

Comment:
Please make use of form validation from bootstrap instead of creating your own error message. Check out [https://getbootstrap.com/docs/5.3/forms/validation/](https://getbootstrap.com/docs/5.3/forms/validation/)